soc/st/stm32/stm32wbx/power.c

```diff
@@ -85,6 +86,19 @@ void pm_state_set(enum pm_state state, uint8_t substate_id)
 			return;
 		}
 
+#if defined(STM32_HSI48_ENABLED)
+		if (IS_ENABLED(STM32_HSI48_ENABLED)) {
```

Comment:
Only required if CLK48 is used. Same conditional statements for where the matching semaphore lock was done. See https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/clock_control/clock_stm32_ll_common.c#L756

Reply:
ah, I think I know why I was confused then :) this should be as follows, right?

```suggestion
#if defined(STM32_SRC_HSI48)
		if (IS_ENABLED(STM32_HSI48_ENABLED)) {
```

Reply:
Technically I suppose there is no harm if these lines are removed. If CLK48 isn't used (RNG or USB not configured) then this would result in a useless instruction (consuming a few bytes of code).

But I chose to copy the design pattern from the existing implementation for consistency and clarity.

Reply:
Yes, that is true, didn't think that far. Thanks for pointing that out. Fixed it now.